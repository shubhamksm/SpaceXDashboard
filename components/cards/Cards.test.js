import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import Cards from './';

describe("Snapshot Test - Cards", () => {
  it('Should display mission details', () => {
    const component = renderer.create(<Cards 
      imgSrc="https://images2.imgbox.com/a0/cb/s1h2RuR0_o.png"
      title="FalconSat #1"
      missionIds={[123456, 123456, 123456]}
      launch_year={2006}
      launch_success={true}
      land_success={true}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Content Check - Cards", () => {
  it('Should have correct details and display', () => {
    const component = shallow(<Cards 
      imgSrc="https://images2.imgbox.com/a0/cb/s1h2RuR0_o.png"
      title="FalconSat #1"
      missionIds={[123456, 123456, 123456]}
      launch_year={2006}
      launch_success={true}
      land_success={true}
    />);

    expect(component.find("h4").text()).toEqual("FalconSat #1");
  });
});