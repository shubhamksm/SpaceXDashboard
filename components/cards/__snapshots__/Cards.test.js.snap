// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test - Cards Should display mission details 1`] = `
<div
  className="card"
>
  <div>
    <div
      className="imgBox"
    >
      <img
        alt="mission picture"
        src="https://images2.imgbox.com/a0/cb/s1h2RuR0_o.png"
      />
    </div>
    <h4>
      FalconSat #1
    </h4>
  </div>
  <div>
    <p>
      <b>
        Mission Ids: 
      </b>
    </p>
    <ul>
      <li>
        123456
      </li>
      <li>
        123456
      </li>
      <li>
        123456
      </li>
    </ul>
    <p>
      <b>
        Launch Year: 
      </b>
      2006
    </p>
    <p>
      <b>
        Successful Launch: 
      </b>
      true
    </p>
    <p>
      <b>
        Successful Landing: 
      </b>
      true
    </p>
  </div>
</div>
`;
